### PR TITLE
Add treeshaking and required tokens

### PR DESCRIPTION
**Pull request by pearofducks**

### CODE REVIEW

**Comment by BalbinaK** [comment]
[Tokenizer: improvements](https://trello.com/c/T58Cy1e1/5-tokenizer-improvements)

File: src/treeshake.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @arg {TokenResult} tokens
```
Comment (BalbinaK) [inline src/treeshake.js]: Is this also an array of TokenResult items?

File: src/treeshake.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * @returns {Array.<TokenResult>}
+ */
+export const treeshake = (tokens) => {
+  const maps = tokens.filter(t => t.isMap)
+  // t.token[0] is the name of the token definition
+  // mappedToken.token[1] is the destination for the mapping
+  const shakenDefs = tokens.filter(t => t.isDef).filter(t => maps.some(mappedToken => mappedToken.token[1] === t.token[0]))
+  return [...shakenDefs, ...maps]
```
Comment (BalbinaK) [inline src/treeshake.js]: I always need a few minutes to figure out what the first and second element of `t.token`/`mappedToken.token` array refers to (despite the comments), so I was thinking to try a more object-based approach here instead. I tried making `t.token` into an object like `token: { name: "border-success", value: "var(--x-blue-50)"}`, so we'd then compare `mappedToken.token.value === t.token.name`. All works correctly with a few adjustments in `toCSSDef` and `toCSSMap` functions, but I'm happy to take you through it to see if that makes sense 😊

File: src/util.js [hunk 3]
```diff
@@ -2,6 +2,18 @@ import fs from 'node:fs'
...
 import slugify from './slugify.js'
 
+/**
+ * @arg {Array.<TokenResult>} tokens
+ * @arg {Array.<string>} requirements
+ */
+export const requireTokens = (tokens, requirements) => {
+  if (requirements?.length) {
+    for (const r of requirements) {
+      if (!tokens.some(t => t.token[0] === r)) throw `Missing required token: '${r}'`
```
Comment (BalbinaK) [inline src/util.js]: Same here, it always makes me stop and think hard what `t.token[0]` was. 😄 So with an object-based approach, we'd then compare requirement with `t.token.name`, which I felt would give some more context when reading this function statement.

File: index.js [hunk 4]
```diff
@@ -11,8 +12,15 @@ export default function (folderPath, options = {}) {
...
+    : tokens
+  try {
+    requireTokens(shakenTokens, options.requirements)
+  } catch (err) {
+    console.error(err, `in directory ${realPath}`)
+    process.exit(1)
+  }
 
   const darkTokens = shakenTokens.filter(t => t.isDark).map(getCSS)
   const normalTokens = shakenTokens.filter(t => !t.isDark).map(getCSS)
```
Comment (BalbinaK) [inline index.js:26]: If the list of tokens that are used for generating css can be shorter due to treeshaking, would it make sense to run `toCSSDef` and `toCSSMap` as part of `getCSS`?

**Review by BalbinaK — COMMENTED** [review]
Whenever I was reviewing this PR I had to console.log things to understand what different functions were returning, and if at some point I had to switch focus(meetings etc.) and came back after a day, it was pretty hard figuring out what was actually happening here, so I always ended up starting reading the code from the beginning 😅 I'm not sure if it's just me or if we could improve readability of this module, but I thought I'd at least check with you some small changes in structure of the tokens arrays:

File: index.js (hunk 4 above)
Comment (pearofducks) [reply index.js:26]: If this was some kind of realtime app, definitely!

Those functions have almost zero performance impact, and this module doesn't need to be performant - so it's not worth investigating/changing IMO.

**Review by BalbinaK — APPROVED** [review]

**Pull request merged**